TheraLang.BLL/Services/UserManagementService.cs

```diff
@@ -1,28 +1,39 @@
 ﻿using Common.Helpers.PasswordHelper;
 using Microsoft.EntityFrameworkCore;
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
 using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TheraLang.BLL.DataTransferObjects;
 using TheraLang.BLL.Interfaces;
 using TheraLang.DAL.Entities;
 using TheraLang.DAL.UnitOfWork;
+using TheraLang.BLL.DataTransferObjects;
+using AutoMapper;
 
 namespace TheraLang.BLL.Services
 {
     public class UserManagementService : IUserManagementService
     {
         private readonly IUnitOfWork _unitOfWork;
+        private readonly IFileService _fileService;
 
-        public UserManagementService(IUnitOfWork unitOfWork)
+        public UserManagementService(IUnitOfWork unitOfWork, IFileService fileService)
         {
             _unitOfWork = unitOfWork;
+            _fileService = fileService;
         }
 
-        public async Task<User> GetUser(string userName, string password)
+
+        public async Task<User> GetUser(string email, string password)
         {
             var user = await _unitOfWork.Repository<User>().GetAll()
                     .Include(x => x.Role)
-                .FirstOrDefaultAsync(u => u.UserName == userName && PasswordHasher.VerifyHashedPassword(u.PasswordHash, password));
-            
+                .FirstOrDefaultAsync(u => u.Email == email && PasswordHasher.VerifyHashedPassword(u.PasswordHash, password));
```

Comment:
Verify password better after getting user from db. Cuz if you pass function to Linq it will get all the users and then filter by that function

Reply:
We have this action only in one request, why do we need two commands and two time access to the db instead of one?

Reply:
```suggestion
                    .FirstOrDefaultAsync(u => u.Email == email);
            if (PasswordHasher.VerifyHashedPassword(user.PasswordHash, password))
            {
                return user;
            }
            else
            {
                throw new Exception("...") // Insert something meaningful
            }
```

Reply:
I'm not agree with you, that it will be better. 
You just add mess in code and all.

Reply:
How do you expect PasswordHasher.VerifyHashedPassword(u.PasswordHash, password) to be converted in sql?

Reply:
Okay, I understoode what you mean

Reply:
done